pkg/operator/ceph/disruption/clusterdisruption/osd.go

```diff
@@ -168,130 +182,174 @@ func (r *ReconcileClusterDisruption) reconcilePDBsForOSDs(
 			logger.Debugf("Ceph %q cluster not ready, cannot check Ceph status yet.", request.Namespace)
 			return nil
 		}
-		return errors.Wrapf(err, "could not check cluster health")
+		return errors.Wrapf(err, "failed to check cluster health")
 	}
+
+	activeDrains := len(drainingFailureDomains) > 0
+
+	// create blocking PDBs for all the OSDs that are not being drained right now
+	if activeDrains {
+		logger.Infof("pg health: %q. detected drains on %q: %v", pgHealthMsg, failureDomain, drainingFailureDomains)
+		for _, failureDomainName := range allFailureDomains {
+			// create blocking PDB for failure domains not currently draining
+			if failureDomainName != drainingFailureDomains[0] {
```

Comment:
Do we track somewhere which failure domain is being drained? The failure domain list may not be ordered, so if the slice is in a different order in a later reconcile, we might unintentionally switch which failure domain is allowed to drain.

Reply:
using sets.List() method ([here](https://github.com/rook/rook/blob/c096888780d5ddd69a9a5f5f8f2a0c842cb8f3e4/pkg/operator/ceph/disruption/clusterdisruption/osd.go#L372)) that  returns the contents as a sorted string slice. Also, I think we won't scenarios where in first reconcile we have draining nodes as [zone-b] and in second reconcile we get [zone-a, zone-b] because we are blocking drains until OSDs are back and PGs are healthy in zone-a